source/backend/api/Services/DocumentFileService.cs

```diff
@@ -89,333 +91,509 @@ public IList<T> GetFileDocuments<T>(FileType fileType, long fileId)
             }
         }
 
-        public async Task<DocumentUploadRelationshipResponse> UploadResearchDocumentAsync(long researchFileId, DocumentUploadRequest uploadRequest)
+        public async Task UploadAcquisitionDocument(long acquisitionFileId, DocumentUploadRequest uploadRequest)
         {
-            Logger.LogInformation("Uploading document for single research file");
-            User.ThrowIfNotAllAuthorized(Permissions.DocumentAdd, Permissions.ResearchFileEdit);
+            Logger.LogInformation("Uploading document for single acquisition file");
+            User.ThrowIfNotAllAuthorized(Permissions.DocumentAdd, Permissions.AcquisitionFileEdit);
+            uploadRequest.ThrowInvalidFileSize();
 
-            // Do not call Mayan if uploaded file is empty (zero-size)
-            ValidateZeroLengthFile(uploadRequest);
+            using var transaction = _documentQueueRepository.BeginTransaction();
 
-            DocumentUploadResponse uploadResult = await documentService.UploadDocumentAsync(uploadRequest);
+            // Step 1 - Save the pimsDocument
+            PimsDocument pimsDocument = CreatePimsDocument(uploadRequest);
+            _documentQueueRepository.SaveChanges();
 
-            DocumentUploadRelationshipResponse relationshipResponse = new DocumentUploadRelationshipResponse()
+            // Step 2 - Create the File Document
+            PimsAcquisitionFileDocument newAcquisitionDocument = new()
             {
-                UploadResponse = uploadResult,
+                AcquisitionFileId = acquisitionFileId,
+                DocumentId = pimsDocument.DocumentId,
             };
+            _acquisitionFileDocumentRepository.AddAcquisition(newAcquisitionDocument);
+            _documentQueueRepository.SaveChanges();
```

Comment:
I don't think you want to call save changes between repository transactions, I'm guessing all of this work should be done in a single transaction to allow rollback if a later transaction fails.

Reply:
this is still a single transaction, SaveChanges() won't commit any work and it will rollback if the transaction fails.

Reply:
at the very least you don't need to call savechanges twice, have a look at that method.

Reply:
not sure what you mean, savechanges is called just 3 times, one for the document, for the file-document and for the que queue document.